plugins/auth-core/src/index.ts

```diff
@@ -272,24 +279,21 @@ class AuthCorePlugin implements AmplicationPlugin {
       staticPath
     );
 
-    staticsInterceptorsFiles.forEach((file) => {
-      staticsFiles.push(file);
-    });
+    await staticsFiles.merge(staticsInterceptorsFiles);
 
     // 1. create user info
     const userInfo = await createUserInfo(context);
     // 2. create token payload interface
     const tokenPayloadInterface = await createTokenPayloadInterface(context);
     // 3. create constants for tests
-    const athConstants = await createAuthConstants(context);
-
-    return [
-      userInfo,
-      tokenPayloadInterface,
-      athConstants,
-      ...staticsFiles,
-      ...modules,
-    ];
+    const authConstants = await createAuthConstants(context);
+
+    await modules.set(userInfo.path, userInfo);
+    await modules.set(tokenPayloadInterface.path, tokenPayloadInterface);
+    await modules.set(authConstants.path, authConstants);
```

Comment:
I noticed that we need need to do this in many places (`.set()`, `.set()`, `.set()`....)
Maybe we can add to the ModuleMap class `setMany()` method.

Something like that:

```
  async setMany(modules: { path: string; module: Module }[]) {
    for (const { path, module } of modules) {
      await this.set(path, module);
    }
  }
```

and then:

```
await modules.setMany([
  { path: "path-1" , module: module-1 },
  { path: "path-2", module: module-2 },
  { path:  "path-3", module: module-3 },
]);

``` 

It probably won't save any lines of code...

Reply:
It doesn't make much sense for me, first, as you've already mentioned, it won't save lines of code, but even more important, it won't improve performance, which is the main reason for bulk operations.